### PR TITLE
authentication with jwt token

### LINKED ISSUE #45: Add Token-based authentication
I see that you can set_authentication using a username and password.  But, in many cases an authentication token is preferred.

### PR DESCRIPTION
**Pull request by driftluo**
close #45

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -69,10 +81,7 @@ impl Client {
         let url = self.build_url("ping", None);
         self.client.get(url).send().map(move |res| {
             if let Ok(res) = res {
-                match res.status().as_u16() {
```
Comment (hpux735) [inline src/client.rs:72]: (again, I'm learning Rust, so sorry if this is annoying...)

What does this change do?  Is it just gardening, or is there a real, and required change in logic?

Comment (driftluo) [reply src/client.rs:72]: It’s just clippy’s suggestion, no actual logical modification

Comment (hpux735) [reply src/client.rs:72]: Awesome, thanks.

**Comment by driftluo** [comment]
why ci https://travis-ci.org/github/driftluo/InfluxDBClient-rs/builds/734672053 doesn't display

**Pull request merged**